docs/environment/spack.md

```diff
@@ -0,0 +1,456 @@
+# Spack: A Package Manager on the UL HPC Platform
+
+[<img width='400px' src='https://cdn.rawgit.com/spack/spack/develop/share/spack/logo/spack-logo-text.svg'/>](https://spack.readthedocs.io/en/latest/#)
+
+
+
+## Introduction to Spack
+
+<p style="text-align: justify;"> Spack is an open-source package manager designed for installing, building, and managing scientific software across a wide range of systems—from personal laptops to the world’s largest supercomputers. It supports multiple versions, compilers, and configurations of packages, all coexisting without conflict. Spack is especially popular in high-performance computing (HPC) environments due to its non-destructive installations, flexible dependency resolution, and robust support for complex software stacks. </p>
+
+### Why spack ? 
+
+<p style="text-align: justify;">Spack simplifies the software installation process in scientific computing. It uses a concise <code>spec</code> syntax to let users define versions, compilers, build options, and dependencies in a readable format. Package recipes in Spack are written in pure Python, allowing contributors to manage many builds with a single file. With over 8,500 official packages available, Spack offers great flexibility—and it's not limited to pre-existing packages. Users can create custom ( e.g. <code>package.py</code> ) files for software not yet available in the Spack pre-defined packages. <a href="https://spack.readthedocs.io/en/latest/" target="_blank">[1]</a> </p>
+
+### Key Features of Spack
+<ul style="text-align: justify;"> 
+
+<li><strong>Multiple Versions & Configurations:</strong> Easily install multiple versions of the same software with different compilers or build options.</li> 
+
+<li><strong>Custom Dependencies:</strong> Flexibly control dependencies, even choosing between alternative implementations.</li> 
+
+<li><strong>Non-destructive Installs:</strong> New installations do not interfere with existing packages.</li> 
+
+<li><strong>Package Coexistence:</strong> Different builds of the same software can live side by side.</li>
+
+ <li><strong>Easy Package Creation:</strong> Write simple ( e.g. <code>package.py</code> ) files in Python to add new software to Spack.</li> 
+ 
+ <li><strong>Virtual Environments:</strong> Create isolated environments for experiments or projects.</li> 
+ 
+ </ul>
+
+
+### Key Resources
+
+Below are essential resources including its official documentation, usage guides, packaging tutorials, and community links:
```

Comment:
Explicitly note that we do not teach user's Spack and refer them to the excellent documentation.